src/wallet/wallet.cpp

```diff
@@ -4632,20 +4639,24 @@ CKeyPool::CKeyPool(const CPubKey& vchPubKeyIn, bool internalIn)
     m_pre_split = false;
 }
 
-void CWalletTx::SetConf(Status status, const uint256& block_hash, int posInBlock)
+void CWalletTx::SetConf(Status status, const uint256& block_hash, int posInBlock, int block_height)
 {
     // Update tx status
     m_confirm.status = status;
 
     // Update the tx's hashBlock
     m_confirm.hashBlock = block_hash;
 
+    // set tx height
+    m_confirm.block_height = block_height;
+
     // set the position of the transaction in the block
     m_confirm.nIndex = posInBlock;
 }
 
 int CWalletTx::GetDepthInMainChain(interfaces::Chain::Lock& locked_chain) const
 {
+    assert(m_confirm.block_height >= -1);
```

Comment:
In commit "Add block_height field in struct Confirmation" (de8fe7d717042823f2aa429be2a260904a10b42e)

This seems harmless, but it's unclear why it's part of this commit. Is this line meant to added to another commit where the block_height is actually used? Also, I don't see where -1 would be set or allowed for this.

Reply:
Removed, was there to be sure block_height was initialized in a previous version, now it has to be with `Confirmation` default initialization